TrackPrivateBase's label/language should return a String

jyavenard opened the pull request:
#### bd2c5cb034caad9a6d36421ecb4acacf4b35ac20
<pre>
TrackPrivateBase&apos;s label/language should return a String
<a href="https://bugs.webkit.org/show_bug.cgi?id=301842">https://bugs.webkit.org/show_bug.cgi?id=301842</a>
<a href="https://rdar.apple.com/163911707">rdar://163911707</a>

Reviewed by Youenn Fablet.

In preparation for running the SourceBufferPrivate/MediaSourcePrivate
in the content process, we need the TrackPrivateBase to use Strings rather
than AtomStrings.
The TrackPrivateBase is created and owned by the SourceBufferPrivate.
While the SourceBufferPrivate was up to now a main thread only object,
this assumption is no longer going to be true: SourceBufferPrivate may
run in a worker&apos;s thread.
By making the TrackPrivateBase use Strings instead of AtomStrings we can
allow to move the objects between threads.
Many of the inherited class members had been moved to Strings in 274956@main.
We converted two base class members.
When building an {Audio,Video,Text}Track from an {Audio,Video,Text}TrackPrivate
take an isolatedCopy before converting to an AtomString. The AtomString conversion
mutate the original String, even if const that makes it non-isolated.

No change in observable behaviours.
Covered by existing tests.
* Source/WebCore/html/track/AudioTrack.cpp:
(WebCore::AudioTrack::labelChanged):
(WebCore::AudioTrack::languageChanged):
* Source/WebCore/html/track/AudioTrack.h:
* Source/WebCore/html/track/InbandTextTrack.cpp:
(WebCore::InbandTextTrack::InbandTextTrack):
(WebCore::InbandTextTrack::inBandMetadataTrackDispatchType const):
(WebCore::InbandTextTrack::labelChanged):
(WebCore::InbandTextTrack::languageChanged):
* Source/WebCore/html/track/InbandTextTrack.h:
* Source/WebCore/html/track/TextTrack.h:
(WebCore::TextTrack::inBandMetadataTrackDispatchType const):
* Source/WebCore/html/track/TrackBase.cpp:
(WebCore::trackUID):
(WebCore::MediaTrackBase::MediaTrackBase):
* Source/WebCore/html/track/TrackBase.h:
* Source/WebCore/html/track/VideoTrack.cpp:
(WebCore::VideoTrack::VideoTrack):
(WebCore::VideoTrack::labelChanged):
(WebCore::VideoTrack::languageChanged):
* Source/WebCore/html/track/VideoTrack.h:
* Source/WebCore/html/track/WebVTTParser.cpp:
(WebCore::WebVTTParser::parseCueData):
* Source/WebCore/platform/graphics/InbandTextTrackPrivate.h:
(WebCore::InbandTextTrackPrivate::inBandMetadataTrackDispatchType const):
* Source/WebCore/platform/graphics/TrackPrivateBase.cpp:
(WebCore::TrackPrivateBase::trackUID const):
* Source/WebCore/platform/graphics/TrackPrivateBase.h:
* Source/WebCore/platform/graphics/TrackPrivateBaseClient.h:
* Source/WebCore/platform/graphics/avfoundation/AVTrackPrivateAVFObjCImpl.h:
* Source/WebCore/platform/graphics/avfoundation/AVTrackPrivateAVFObjCImpl.mm:
(WebCore::AVTrackPrivateAVFObjCImpl::label const):
(WebCore::AVTrackPrivateAVFObjCImpl::language const):
* Source/WebCore/platform/graphics/avfoundation/AudioTrackPrivateAVF.h:
(WebCore::AudioTrackPrivateAVF::setLabel):
(WebCore::AudioTrackPrivateAVF::setLanguage):
* Source/WebCore/platform/graphics/avfoundation/InbandMetadataTextTrackPrivateAVF.h:
(WebCore::InbandMetadataTextTrackPrivateAVF::setInBandMetadataTrackDispatchType):
* Source/WebCore/platform/graphics/avfoundation/InbandTextTrackPrivateAVF.cpp:
(WebCore::InbandTextTrackPrivateAVF::processVTTSample):
* Source/WebCore/platform/graphics/avfoundation/InbandTextTrackPrivateAVF.h:
(WebCore::InbandTextTrackPrivateAVF::setLabel):
(WebCore::InbandTextTrackPrivateAVF::setLanguage):
* Source/WebCore/platform/graphics/avfoundation/VideoTrackPrivateAVF.h:
(WebCore::VideoTrackPrivateAVF::setLabel):
(WebCore::VideoTrackPrivateAVF::setLanguage):
* Source/WebCore/platform/graphics/avfoundation/objc/InbandChapterTrackPrivateAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/InbandChapterTrackPrivateAVFObjC.mm:
(WebCore::InbandChapterTrackPrivateAVFObjC::processChapters):
(WebCore::InbandChapterTrackPrivateAVFObjC::language const):
* Source/WebCore/platform/graphics/avfoundation/objc/InbandTextTrackPrivateAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/InbandTextTrackPrivateAVFObjC.mm:
(WebCore::InbandTextTrackPrivateAVFObjC::label const):
(WebCore::InbandTextTrackPrivateAVFObjC::language const):
* Source/WebCore/platform/graphics/avfoundation/objc/VideoTrackPrivateMediaSourceAVFObjC.mm:
(WebCore::VideoTrackPrivateMediaSourceAVFObjC::resetPropertiesFromTrack):
* Source/WebCore/platform/graphics/cocoa/AudioTrackPrivateWebM.cpp:
(WebCore::AudioTrackPrivateWebM::label const):
(WebCore::AudioTrackPrivateWebM::language const):
* Source/WebCore/platform/graphics/cocoa/AudioTrackPrivateWebM.h:
* Source/WebCore/platform/graphics/cocoa/VideoTrackPrivateWebM.cpp:
(WebCore::VideoTrackPrivateWebM::label const):
(WebCore::VideoTrackPrivateWebM::language const):
* Source/WebCore/platform/graphics/cocoa/VideoTrackPrivateWebM.h:
* Source/WebCore/platform/graphics/gstreamer/AudioTrackPrivateGStreamer.h:
* Source/WebCore/platform/graphics/gstreamer/TrackPrivateBaseGStreamer.cpp:
(WebCore::TrackPrivateBaseGStreamer::TrackPrivateBaseGStreamer):
(WebCore::TrackPrivateBaseGStreamer::setPad):
(WebCore::TrackPrivateBaseGStreamer::getLanguageCode):
(WebCore::TrackPrivateBaseGStreamer::notifyTrackOfTagsChanged):
(WebCore::TrackPrivateBaseGStreamer::notifyTrackOfStreamChanged):
(WebCore::TrackPrivateBaseGStreamer::updateTrackIDFromTags):
* Source/WebCore/platform/graphics/gstreamer/TrackPrivateBaseGStreamer.h:
(WebCore::TrackPrivateBaseGStreamer::gstStreamId const):
* Source/WebCore/platform/graphics/gstreamer/VideoTrackPrivateGStreamer.h:
* Source/WebCore/platform/graphics/iso/ISOVTTCue.cpp:
(WebCore::ISOWebVTTCue::ISOWebVTTCue):
(WebCore::ISOWebVTTCue::parse):
(WebCore::ISOWebVTTCue::isolatedCopy const):
(WebCore::ISOWebVTTCue::isolatedCopy):
* Source/WebCore/platform/graphics/iso/ISOVTTCue.h:
* Source/WebCore/platform/mediastream/AudioTrackPrivateMediaStream.h:
* Source/WebCore/platform/mediastream/VideoTrackPrivateMediaStream.h:
* Source/WebKit/GPUProcess/media/RemoteAudioTrackProxy.cpp:
(WebKit::RemoteAudioTrackProxy::labelChanged):
(WebKit::RemoteAudioTrackProxy::languageChanged):
* Source/WebKit/GPUProcess/media/RemoteAudioTrackProxy.h:
* Source/WebKit/GPUProcess/media/RemoteTextTrackProxy.cpp:
(WebKit::RemoteTextTrackProxy::labelChanged):
(WebKit::RemoteTextTrackProxy::languageChanged):
* Source/WebKit/GPUProcess/media/RemoteTextTrackProxy.h:
* Source/WebKit/GPUProcess/media/RemoteVideoTrackProxy.cpp:
(WebKit::RemoteVideoTrackProxy::labelChanged):
(WebKit::RemoteVideoTrackProxy::languageChanged):
* Source/WebKit/GPUProcess/media/RemoteVideoTrackProxy.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/WebProcess/GPU/media/AudioTrackPrivateRemote.h:
* Source/WebKit/WebProcess/GPU/media/TextTrackPrivateRemote.h:
* Source/WebKit/WebProcess/GPU/media/VideoTrackPrivateRemote.h:

Canonical link: <a href="https://commits.webkit.org/302527@main">https://commits.webkit.org/302527@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3eec9b8691a2a95c4ace59798737fc230be42f73

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/129385 "5 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/1642 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/40224 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/136762 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/80793 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/6f5b46f7-b777-44fe-83f2-1158e40eb5d0) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/131256 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/1572 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/1518 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/98539 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/66434 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/2c45ae95-cf6c-4ffc-9335-beb7b451853e) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/132332 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/1228 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/115886 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/79190 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/7a20c2d0-a797-4059-a8f1-bddc4ac10f5b) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/1150 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/34018 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/80039 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/109603 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/34518 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/139235 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/1432 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/1381 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/107065 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/1474 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/112231 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/106909 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27216 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/1171 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/30751 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/54090 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/1503 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/64866 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/1320 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/1357 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/1425 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebCore/html/track/InbandTextTrack.cpp:
```diff
@@ -60,7 +60,7 @@ Ref<InbandTextTrack> InbandTextTrack::create(ScriptExecutionContext& context, In
 
 InbandTextTrack::InbandTextTrack(ScriptExecutionContext& context, InbandTextTrackPrivate& trackPrivate)
-    : TextTrack(&context, emptyAtom(), trackPrivate.id(), trackPrivate.label(), trackPrivate.language(), InBand)
+: TextTrack(&context, emptyAtom(), trackPrivate.id(), AtomString { trackPrivate.label() }, AtomString { trackPrivate.language() }, InBand)
```
Fix the indentation

youennf commented on Source/WebCore/html/track/AudioTrack.h:
```diff
@@ -82,8 +82,8 @@ class AudioTrack final : public MediaTrackBase, private AudioTrackPrivateClient
     void idChanged(TrackID) final;
-    void labelChanged(const AtomString&) final;
-    void languageChanged(const AtomString&) final;
+    void labelChanged(const String&) final;
```
Do we need to atomise label?

jyavenard replied:
it is atomised in the implementation.

youennf commented on Source/WebCore/html/track/WebVTTParser.cpp:
```diff
@@ -145,7 +145,7 @@ void WebVTTParser::parseCueData(const ISOWebVTTCue& data)
 
     cue->setContent(data.cueText());
-    cue->setId(data.id());
+    cue->setId(AtomString { data.id() });
```
Ditto for id, is it worth using AtomString?

jyavenard replied:
I only looked at changing the TrackPrivate which can be created and controlled on a different thread.
AudioTrack/VideoTrack/TextTrack (which are TrackPrivateClient) are main-thread only DOM objects.

The change would be much bigger if I were to change those too.

youennf commented on Source/WebKit/WebProcess/GPU/media/TextTrackPrivateRemote.h:
```diff
@@ -75,10 +75,10 @@ class TextTrackPrivateRemote final : public WebCore::InbandTextTrackPrivate {
+    String language() const final { return m_language; }
     int trackIndex() const final { return m_trackIndex; }
-    AtomString inBandMetadataTrackDispatchType() const final { return AtomString { m_inBandMetadataTrackDispatchType.isolatedCopy() }; }
+    String inBandMetadataTrackDispatchType() const final { return AtomString { m_inBandMetadataTrackDispatchType.isolatedCopy() }; }
```
Why are we doing differently than for m_label and m_language? Why do we need to call `AtomString` constructor?
Also, in some cases, we are adding isolatedCopy() and in some cases, we are removing them.
This is hard to follow.

jyavenard replied:
it's a remnant of MSE in a worker being done in the GPU process.
The MediaSource/SourceBuffer can be in a worker, but with the player in the GPU process, the TrackPrivates are created by the MediaPlayerPrivateRemote on the main thread.
So the TrackPrivate in this case is constructed on the main thread to be given back to the SourceBuffer in the worker; it needed isolatedCopy() then.

With the player in the web content process, the track is created by the SourceBufferPrivate directly and do not need this main thread jump.

I should have removed that AtomString constructor it's not needed. I'll do so.

Also, apparently StringImpl are now thread-safe so isolatedCopy will become a no-op very soon.

jyavenard replied:
> Why are we doing differently than for m_label and m_language? Why do we need to call `AtomString` constructor? Also, in some cases, we are adding isolatedCopy() and in some cases, we are removing them. This is hard to follow.

oh I remember why I did it here a couple of years ago. When you atomised the string, it's self-modifying. so it needs to be isolated again.

I think it will be safer to remove all AtomString everywhere here.

jyavenard replied:
I've re-visited. I've moved the responsibility of taking an isolatedCopy to the TrackPrivateClient that will be transforming the string to an AtomString. makes more sense that way

jyavenard committed:
TrackPrivateBase's label/language should return a String
https://bugs.webkit.org/show_bug.cgi?id=301842
rdar://163911707

Reviewed by Youenn Fablet.

In preparation for running the SourceBufferPrivate/MediaSourcePrivate
in the content process, we need the TrackPrivateBase to use Strings rather
than AtomStrings.
The TrackPrivateBase is created and owned by the SourceBufferPrivate.
While the SourceBufferPrivate was up to now a main thread only object,
this assumption is no longer going to be true: SourceBufferPrivate may
run in a worker's thread.
By making the TrackPrivateBase use Strings instead of AtomStrings we can
allow to move the objects between threads.
Many of the inherited class members had been moved to Strings in 274956@main.
We converted two base class members.
When building an {Audio,Video,Text}Track from an {Audio,Video,Text}TrackPrivate
take an isolatedCopy before converting to an AtomString. The AtomString conversion
mutate the original String, even if const that makes it non-isolated.

No change in observable behaviours.
Covered by existing tests.
* Source/WebCore/html/track/AudioTrack.cpp:
(WebCore::AudioTrack::labelChanged):
(WebCore::AudioTrack::languageChanged):
* Source/WebCore/html/track/AudioTrack.h:
* Source/WebCore/html/track/InbandTextTrack.cpp:
(WebCore::InbandTextTrack::InbandTextTrack):
(WebCore::InbandTextTrack::inBandMetadataTrackDispatchType const):
(WebCore::InbandTextTrack::labelChanged):
(WebCore::InbandTextTrack::languageChanged):
* Source/WebCore/html/track/InbandTextTrack.h:
* Source/WebCore/html/track/TextTrack.h:
(WebCore::TextTrack::inBandMetadataTrackDispatchType const):
* Source/WebCore/html/track/TrackBase.cpp:
(WebCore::trackUID):
(WebCore::MediaTrackBase::MediaTrackBase):
* Source/WebCore/html/track/TrackBase.h:
* Source/WebCore/html/track/VideoTrack.cpp:
(WebCore::VideoTrack::VideoTrack):
(WebCore::VideoTrack::labelChanged):
(WebCore::VideoTrack::languageChanged):
* Source/WebCore/html/track/VideoTrack.h:
* Source/WebCore/html/track/WebVTTParser.cpp:
(WebCore::WebVTTParser::parseCueData):
* Source/WebCore/platform/graphics/InbandTextTrackPrivate.h:
(WebCore::InbandTextTrackPrivate::inBandMetadataTrackDispatchType const):
* Source/WebCore/platform/graphics/TrackPrivateBase.cpp:
(WebCore::TrackPrivateBase::trackUID const):
* Source/WebCore/platform/graphics/TrackPrivateBase.h:
* Source/WebCore/platform/graphics/TrackPrivateBaseClient.h:
* Source/WebCore/platform/graphics/avfoundation/AVTrackPrivateAVFObjCImpl.h:
* Source/WebCore/platform/graphics/avfoundation/AVTrackPrivateAVFObjCImpl.mm:
(WebCore::AVTrackPrivateAVFObjCImpl::label const):
(WebCore::AVTrackPrivateAVFObjCImpl::language const):
* Source/WebCore/platform/graphics/avfoundation/AudioTrackPrivateAVF.h:
(WebCore::AudioTrackPrivateAVF::setLabel):
(WebCore::AudioTrackPrivateAVF::setLanguage):
* Source/WebCore/platform/graphics/avfoundation/InbandMetadataTextTrackPrivateAVF.h:
(WebCore::InbandMetadataTextTrackPrivateAVF::setInBandMetadataTrackDispatchType):
* Source/WebCore/platform/graphics/avfoundation/InbandTextTrackPrivateAVF.cpp:
(WebCore::InbandTextTrackPrivateAVF::processVTTSample):
* Source/WebCore/platform/graphics/avfoundation/InbandTextTrackPrivateAVF.h:
(WebCore::InbandTextTrackPrivateAVF::setLabel):
(WebCore::InbandTextTrackPrivateAVF::setLanguage):
* Source/WebCore/platform/graphics/avfoundation/VideoTrackPrivateAVF.h:
(WebCore::VideoTrackPrivateAVF::setLabel):
(WebCore::VideoTrackPrivateAVF::setLanguage):
* Source/WebCore/platform/graphics/avfoundation/objc/InbandChapterTrackPrivateAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/InbandChapterTrackPrivateAVFObjC.mm:
(WebCore::InbandChapterTrackPrivateAVFObjC::processChapters):
(WebCore::InbandChapterTrackPrivateAVFObjC::language const):
* Source/WebCore/platform/graphics/avfoundation/objc/InbandTextTrackPrivateAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/InbandTextTrackPrivateAVFObjC.mm:
(WebCore::InbandTextTrackPrivateAVFObjC::label const):
(WebCore::InbandTextTrackPrivateAVFObjC::language const):
* Source/WebCore/platform/graphics/avfoundation/objc/VideoTrackPrivateMediaSourceAVFObjC.mm:
(WebCore::VideoTrackPrivateMediaSourceAVFObjC::resetPropertiesFromTrack):
* Source/WebCore/platform/graphics/cocoa/AudioTrackPrivateWebM.cpp:
(WebCore::AudioTrackPrivateWebM::label const):
(WebCore::AudioTrackPrivateWebM::language const):
* Source/WebCore/platform/graphics/cocoa/AudioTrackPrivateWebM.h:
* Source/WebCore/platform/graphics/cocoa/VideoTrackPrivateWebM.cpp:
(WebCore::VideoTrackPrivateWebM::label const):
(WebCore::VideoTrackPrivateWebM::language const):
* Source/WebCore/platform/graphics/cocoa/VideoTrackPrivateWebM.h:
* Source/WebCore/platform/graphics/gstreamer/AudioTrackPrivateGStreamer.h:
* Source/WebCore/platform/graphics/gstreamer/TrackPrivateBaseGStreamer.cpp:
(WebCore::TrackPrivateBaseGStreamer::TrackPrivateBaseGStreamer):
(WebCore::TrackPrivateBaseGStreamer::setPad):
(WebCore::TrackPrivateBaseGStreamer::getLanguageCode):
(WebCore::TrackPrivateBaseGStreamer::notifyTrackOfTagsChanged):
(WebCore::TrackPrivateBaseGStreamer::notifyTrackOfStreamChanged):
(WebCore::TrackPrivateBaseGStreamer::updateTrackIDFromTags):
* Source/WebCore/platform/graphics/gstreamer/TrackPrivateBaseGStreamer.h:
(WebCore::TrackPrivateBaseGStreamer::gstStreamId const):
* Source/WebCore/platform/graphics/gstreamer/VideoTrackPrivateGStreamer.h:
* Source/WebCore/platform/graphics/iso/ISOVTTCue.cpp:
(WebCore::ISOWebVTTCue::ISOWebVTTCue):
(WebCore::ISOWebVTTCue::parse):
(WebCore::ISOWebVTTCue::isolatedCopy const):
(WebCore::ISOWebVTTCue::isolatedCopy):
* Source/WebCore/platform/graphics/iso/ISOVTTCue.h:
* Source/WebCore/platform/mediastream/AudioTrackPrivateMediaStream.h:
* Source/WebCore/platform/mediastream/VideoTrackPrivateMediaStream.h:
* Source/WebKit/GPUProcess/media/RemoteAudioTrackProxy.cpp:
(WebKit::RemoteAudioTrackProxy::labelChanged):
(WebKit::RemoteAudioTrackProxy::languageChanged):
* Source/WebKit/GPUProcess/media/RemoteAudioTrackProxy.h:
* Source/WebKit/GPUProcess/media/RemoteTextTrackProxy.cpp:
(WebKit::RemoteTextTrackProxy::labelChanged):
(WebKit::RemoteTextTrackProxy::languageChanged):
* Source/WebKit/GPUProcess/media/RemoteTextTrackProxy.h:
* Source/WebKit/GPUProcess/media/RemoteVideoTrackProxy.cpp:
(WebKit::RemoteVideoTrackProxy::labelChanged):
(WebKit::RemoteVideoTrackProxy::languageChanged):
* Source/WebKit/GPUProcess/media/RemoteVideoTrackProxy.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/WebProcess/GPU/media/AudioTrackPrivateRemote.h:
* Source/WebKit/WebProcess/GPU/media/TextTrackPrivateRemote.h:
* Source/WebKit/WebProcess/GPU/media/VideoTrackPrivateRemote.h:

Canonical link: https://commits.webkit.org/302527@main
